packages/ui-inputs/src/search-input.tsx

```diff
@@ -0,0 +1,146 @@
+}
+
+const useStyles = createUseStyles((theme: Theme) => ({
+    clearBtn: {
+        cursor: 'pointer',
+        color: theme.colors.grayscale3,
+        transform: 'rotate(45deg)',
+    },
+}), { theming });
+
+const resolveOnChange = (
+    target: InputElement,
+    e:
+        | React.ChangeEvent<InputElement>
+        | React.MouseEvent<HTMLElement, MouseEvent>,
+    onChange?: (event: React.ChangeEvent<InputElement>) => void,
+): void => {
+    const callOnChange = (event: React.ChangeEvent<InputElement>) => {
+        if (typeof onChange === 'function') {
+            onChange(event);
+        }
+    };
+    if (target) {
+        let event = e;
+        if (e.type === 'click') {
+            // click clear icon
+            event = Object.create(e);
+            event.target = target;
+            event.currentTarget = target;
+            const originalInputValue = target.value;
+            // change target ref value cause e.target.value should be '' when clear input
+            target.value = '';
+            callOnChange(event as React.ChangeEvent<InputElement>);
+            // reset target ref value
+            target.value = originalInputValue;
+            return;
+        }
+        callOnChange(event as React.ChangeEvent<InputElement>);
+    }
+};
+
+/**
+ * Search input is an extension of TextInput component that provides additional ability to clear a user's input by clicking the clear button in the input.
+ * The clear button will appear once user input some value in the SearchInput.
+ */
+export const SearchInput = (props: SearchInputFieldProps) => {
+    const { value, defaultValue, disabled, placeholder, inputRef, hasError, isBusy, onChange } = props;
+    const classes = useStyles(props);
+    const clearBtnClass = classNames({ [classes.clearBtn]: true });
+    const [localValue, setLocalValue] = useState<string | undefined>('');
+    const [showClear, setShowClear] = useState(false);
+    const [inputEl, setInputEl] = useState<InputElement>(null);
+    const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        setLocalValue(event.target.value);
+        resolveOnChange(inputEl, event, onChange);
+    };
+    const clearInput = (event: React.MouseEvent<HTMLElement, MouseEvent>) => {
+        setLocalValue('');
+        inputEl?.focus();
+        resolveOnChange(inputEl, event, onChange);
+    };
+    
+    const saveInput = (el: InputElement) => {
+        setInputEl(el);
+        
+        if (!inputRef) {
+            return;
+        }
+        
+        if (typeof inputRef === 'function') {
+            inputRef(el);
+        } else {
+            inputRef.current = el;
+        }
+    };
+    
+    useEffect(() => {
+        setLocalValue(value ?? defaultValue);
+    }, [value, defaultValue]);
+    
+    useEffect(() => {
+        setShowClear(!!localValue);
+    }, [localValue]);
+    
+    return <TextInput value={localValue}
+                      defaultValue={defaultValue}
```

Comment:
since you always provide `localValue`, essentially the underline component is controlled component and you don't really need to pass `defaultValue`